cumulusci/tasks/salesforce/custom_settings.py

```diff
@@ -0,0 +1,135 @@
+import os
+import yaml
+
+from cumulusci.tasks.salesforce import BaseSalesforceApiTask
+from cumulusci.utils import os_friendly_path
+from cumulusci.core.exceptions import TaskOptionsError, CumulusCIException
+
+
+class LoadCustomSettings(BaseSalesforceApiTask):
+    """Load Custom Settings (both List and Hierarchy) into an org
+    from a YAML-format settings file.
+
+    Each top-level YAML key should be the API name of a Custom Setting.
+    List Custom Settings should contain a nested map of names to values.
+    Hierarchy Custom settings should contain a list, each of which contains
+    a `data` key and a `location` key. The `location` key may contain either
+    `profile: <profile name>`, `user: name: <username>`, `user: email: <email>`,
+    or `org`. Example:
+
+    List__c:
+        Test:
+            MyField__c: 1
+        Test 2:
+            MyField__c: 2
+    Hierarchy__c:
+        -
+            location: org
+            data:
+                MyField__c: 1
+        -
+            location:
+                user:
+                    name: test@example.com
+            data:
+                MyField__c: 2"""
+
+    task_options = {
+        "settings_path": {
+            "description": "The path to a YAML settings file",
+            "required": True,
+        }
+    }
+
+    def _init_options(self, kwargs):
+        super()._init_options(kwargs)
+        self.options["settings_path"] = os_friendly_path(
+            self.options.get("settings_path")
+        )
+        if self.options["settings_path"] is None or not os.path.isfile(
+            self.options["settings_path"]
+        ):
+            raise TaskOptionsError(
+                f"File {self.options['settings_path']} does not exist"
+            )
+
+    def _run_task(self):
+        with open(self.options["settings_path"], "r") as f:
+            self.settings = yaml.safe_load(f)
+
+        self.logger.info("Starting Custom Settings load")
+        self._load_settings()
+        self.logger.info("Finished Custom Settings load")
+
+    def _load_settings(self):
+        # For each top-level heading in our YAML doc, create one or more
+        # custom settings.
+
+        for custom_setting, settings_data in self.settings.items():
+            proxy_obj = getattr(self.sf, custom_setting)
+            # If this level is a dict, we're working with a List Custom Setting
+            # If it's a list, we have a Hierarchy Custom Setting.
+            if isinstance(settings_data, dict):
```

Comment:
I would be inclined to have a method for each branch.

And an else clause that raises an error about the YAML format.

Reply:
I added another branch and a new test case to snag bad YAML.